engineering_tools/rosbag_matching/rosbag_match

```diff
@@ -0,0 +1,66 @@
+
+#!/bin/bash 
+
+rosbag=$1
+logfile=$2
+
+# get the bag time from rosbag info
+while read -r line ; do
+
+	if [[ $line =~ \([0-9]+\.[0-9]+\) ]]; then
+		IFS=" "
+		read -ra bagtime <<< "$line"
+		ros_time=$(echo "${bagtime[5]:1:13}" | bc -l)
+		break
+	fi
+
+done < <(rosbag info $rosbag) 
+
+
+# go inside logfiles
+cd $logfile
+
+
+match="0"
+# go through every log directory in logfiles
+for dir in */; do
+	cd $dir
+	echo Log Directory:
+	echo $dir
+	while read -r line ; do
```

Comment:
It is possible that users may run this (by accident, or lack of familiarity, or by choice) in a dir that has other subdirs in it that do not contain logs at all.  Therefore, it would be good to check if the rosout.log file exists before opening it to read (before this while loop).  If none exists, just skip this dir (maybe silently, before printing its name).